### PR TITLE
Stop using the `frontend-maven-plugin` to build Node.js projects

### LINKED ISSUE #24571: Parallel builds stopped working
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Using parallel build to compile Keycloak has stopped working between 21.0.0 and 22.0.0 releases.

### Version

main

### Expected behavior

Compilation should succeed.

### Actual behavior

Compiling fails with errors

```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-js-parent: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-js-admin-client: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-account-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-admin-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
```

### How to Reproduce?

Compile Keycloak by running

```
mvn -T4C clean install -DskipTestsuite -DskipExamples -DskipTests
```

### Anything else?

I've used `git bisect` to track the commit which makes parallel builds fail and it seems to be #17401 and commit  https://github.com/keycloak/keycloak/commit/6d2e57f93a297360c4ebb6082dcd7fc87ccf2448.

### LINKED ISSUE #24781: Do not use `maven.multiModuleProjectDirectory` in build
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Recently use of maven variable `maven.multiModuleProjectDirectory` was introduced in `pom.xml`'s. It is utilized in Javascript build ([link](https://github.com/search?q=repo%3Akeycloak%2Fkeycloak%20maven.multiModuleProjectDirectory&type=code)).

In the code there is warning https://github.com/keycloak/keycloak/blob/07a3def4a27da4a0f6d55feb51db89b6f9c91653/pom.xml#L1895-L1896

The variable is set only during maven build and it is unset if building e.g. in vscode / eclipse. They use different implementation of maven which does not set the variable to project root. 

### Version

main

### Expected behavior

The build should avoid use of maven specific variables that are not supported by IDEs.

### Actual behavior

IDE will run the build with empty install directory, which is interpreted as current directory in each submodule. The files are installed in wrong directories and application cannot be started.

### How to Reproduce?

Open the code in IDE without `maven.multiModuleProjectDirectory` defined. Wait for the project to compile and try to run under debugger.

### Anything else?

Note that it is not possible to work around this problem by executing maven build on command line before build inside IDE, since opening the project in vscode will trigger the `clean` build phase and therefore the previously correctly installed files will be wiped out when opening vscode.

### LINKED ISSUE #25497: Could not download Node.js for Windows runners
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

Regular failures in CI testing for Windows:

```
Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:install-node-and-pnpm (default) on project keycloak-js-adapter-jar: Could not download Node.js from: https://nodejs.org/dist/v18.18.2/win-x64/node.exe: Could not download https://nodejs.org/dist/v18.18.2/win-x64/node.exe: Premature end of Content-Length delimited message body (expected: 71,409,304; received: 38,420,172) -> [Help 1]
```

### Version

main

### Expected behavior

Works everytime

### Actual behavior

Fails fairly frequently

### How to Reproduce?

See GH Actions

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jonkoops**
Removes the `frontend-maven-plugin` dependency from the project and replaces it with direct execution of system Node.js and PNPM binaries through `exec-maven-plugin` . This addresses several issues, but first and foremost build stability on CI, caused by repeated downloads of Node.js by the frontend plugin, which fail often.

This change will **require** users to install Node.js on their system, as this is no longer managed by the frontend plugin, the documentation for building Keycloak has been updated to reflect this.

Closes #25497
Closes #24571
Closes #24781

### CODE REVIEW

File: js/libs/keycloak-admin-client/pom.xml [hunk 1]
```diff
@@ -43,41 +43,34 @@
...
+                        <id>pnpm-pack</id>
+                        <phase>generate-resources</phase>
                         <goals>
-                            <goal>pnpm</goal>
+                            <goal>exec</goal>
                         </goals>
                         <configuration>
-                            <arguments>pack --pack-destination=target</arguments>
-                            <pnpmInheritsProxyConfigFromMaven>false</pnpmInheritsProxyConfigFromMaven>
+                            <executable>pnpm</executable>
```
Comment (jonkoops) [inline js/libs/keycloak-admin-client/pom.xml:69]: Was thinking we could perhaps do this only when the `distribution` profile is enabled to reduce build time, but left it out for now.

File: js/package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "typescript": "^5.3.3",
     "wireit": "^0.14.3"
   },
-  "packageManager": "pnpm@8.10.0",
+  "packageManager": "pnpm@8.14.3",
```
Comment (jonkoops) [inline js/package.json:28]: Bumped the PNPM version we require to the latest stable while we're at it.

**Review by edewit — APPROVED** [review]

**Comment by sschu** [comment]
@jonkoops Have you seen the discussion on the mailing list around this? There were some arguments for keeping it, see https://groups.google.com/g/keycloak-dev/c/OgGTQALxRz8/m/zMtvs_h4AAAJ

**Comment by tsaarni** [comment]
Unfortunately it seems this will not fix parallel builds #24571: running `mvn -T4C clean install -DskipTestsuite -DskipExamples -DskipTests` fails.  

I think the underlying problem of running `pnpm` multiple times in parallel over single shared workspace remains? That is, during parallel build of maven submodules we end up running `pnpm install` (now via `exec`) over shared workspace `js/node_modules/.pnpm/` directory. This use case `pnpm` does not support, it does not have locking or launch single daemon process to dispatch all installs to.  Instead they seemed to recommend running install once recursively from the root dir (to recursively install all `package.json`s in sub directories).

**Comment by jonkoops** [comment]
@sschu Yeah, I'm involved in the discussions, this PR is just to illustrate how this would work without the frontend plugin. We won't merge this until we resolve the discussion.

**Comment by stianst** [comment]
Closing this as sadly (others) didn't want to remove the frontend-maven-plugin. Personally I still think it would be nice to just get rid of it, but hey ;)

**Pull request closed without merging**